src/wallet/wallet.cpp

```diff
@@ -1668,9 +1668,8 @@ set<uint256> CWallet::GetConflicts(const uint256& txid) const
         }
     }
 
-    for (uint32_t i = 0; i < wtx.GetOrchardBundle().GetNumActions(); i++) {
-        OrchardOutPoint op(wtx.GetHash(), i);
-        auto potential_spends = orchardWallet.GetPotentialSpends(op);
+    for (const uint256& nullifier : wtx.GetOrchardBundle().GetNullifiers()) {
+        auto potential_spends = orchardWallet.GetPotentialSpendsFromNullifier(nullifier);
```

Comment:
Given this change, do we need to keep `GetPotentialSpends`? It's not clear to me why the previous implementation wasn't sufficient here, and I think that if we're going to keep both then the difference in how they work should be documented.

Reply:
We have a transaction with nullifiers and outputs. What we want is to find conflicts with the nullifiers _revealed_ in this transaction. `GetPotentialSpends` takes an outpoint, so the initial impl of this PR was instead finding conflicts on the nullifiers _created_ in this transaction. We have no API to look up an outpoint given a nullifier, and adding such an API would just make things way more complex, vs the new API that does the direct lookup.

For the other usage in `IsSpent`, we do care about the nullifiers _created_ in this transaction, so if we didn't have `GetPotentialSpends` then we'd need an API to look up the nullifier for an outpoint. So we'd be removing an API and adding another API, which isn't something we need to do for 4.7.0.